src/ngx_http_lua_subrequest.c

```diff
@@ -382,15 +404,62 @@ ngx_http_lua_ngx_location_capture_multi(lua_State *L)
 
             type = lua_type(L, -1);
 
-            if (type == LUA_TNIL) {
-                method = NGX_HTTP_GET;
+            switch (type) {
+                case LUA_TNIL:
+                    method = NGX_HTTP_GET;
+                    break;
+                case LUA_TNUMBER:
+                    method = (ngx_uint_t) lua_tonumber(L, -1);
```

Comment:
@agentzh why is there a need for a bounds check here? It does not exist in the original code, and later own the method number given here is verified (in the adjust_subrequest function).